### PR TITLE
fix(notifier): fetches own offers without context and limits

### PR DESCRIPTION
**Pull request by itofarina**
## Description
fetches own offers without context and limits
shows spinner while fetching them
avoids refetches and re-renders by removing the context dependency

## Demo

https://user-images.githubusercontent.com/8449567/128037534-a4992f51-1b10-4c92-a90a-9413e6f79b17.mov

### CODE REVIEW

File: src/components/pages/notifier/myoffers/NotifierMyOffersPage.tsx [hunk 1]
```diff
@@ -88,42 +80,32 @@ const NotifierMyOffersPage: FC = () => {
...
+        setMyOffers(offers)
 
-    if (offers.length) {
-      const offer = offers[0]
-      setMyProfile({
-        address: offer.provider,
-        url: offer.url,
-      })
+        if (offers.length) {
+          const offer = offers[0]
```
Comment (jurajpiar) [inline src/components/pages/notifier/myoffers/NotifierMyOffersPage.tsx]:
```suggestion
          const [offer] = offers
```

File: src/components/pages/notifier/myoffers/NotifierMyOffersPage.tsx [hunk 2]
```diff
@@ -88,42 +80,32 @@ const NotifierMyOffersPage: FC = () => {
...
+  const [isLoadingOffers, setIsLoadingOffers] = useState(true)
   const reportError = useErrorReporter()
 
   // Set provider upon wallet connection
-  const myOffers = useMemo(() => {
-    const offers = items
-      .filter(({ provider }) => provider === account)
+  useEffect(() => {
+    const fetchOffers = async (): Promise<void> => {
+      if (offersApi && !offersApi.service) {
```
Comment (jurajpiar) [inline src/components/pages/notifier/myoffers/NotifierMyOffersPage.tsx]: Have a feeling the `offersApi` is always defined, isn't it?

**Review by jurajpiar — APPROVED** [review]
Looking good

File: src/components/pages/notifier/myoffers/NotifierMyOffersPage.tsx (hunk 2 above)
Comment (itofarina) [reply src/components/pages/notifier/myoffers/NotifierMyOffersPage.tsx]: Right, thanks!

**Pull request merged**